server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -80,17 +80,18 @@ public interface BlobContainer {
     InputStream readBlob(String blobName) throws IOException;
 
     /**
-     * Creates a new {@link FetchBlobResult} for the given blob name.
+     * Creates a new {@link InputStreamWithMetadata} for the given blob name.
      *
      * @param   blobName
-     *          The name of the blob to get an {@link InputStream} for.
-     * @return  The {@link FetchBlobResult} of the blob.
+     *          The name of the blob to get an {@link InputStreamWithMetadata} for.
+     * @return  The {@link InputStreamWithMetadata} of the blob.
      * @throws  NoSuchFileException if the blob does not exist
      * @throws  IOException if the blob can not be read.
      */
     @ExperimentalApi
-    default FetchBlobResult readBlobWithMetadata(String blobName) throws IOException {
-        throw new UnsupportedOperationException("readBlobWithMetadata is not implemented yet");
+    default InputStreamWithMetadata readBlobWithMetadata(String blobName) throws IOException {
+        InputStream inputStream = readBlob(blobName);
+        return new InputStreamWithMetadata(inputStream, null);
```

Comment:
Are we changing to make sure that we dont mislead provided a method and not providing the feature it is supposed?

Reply:
Making this function's implementation empty and handling this in transferService implementation